native/src/gpxMultiSegmentsApproximation.h

```diff
@@ -0,0 +1,93 @@
+#ifndef _OSMAND_GPX_MULTI_SEGMENTS_APPROXIMATION_H
+#define _OSMAND_GPX_MULTI_SEGMENTS_APPROXIMATION_H
+
+#include <functional>
+#include <queue>
+
+#include "CommonCollections.h"
+#include "commonOsmAndCore.h"
+
+struct RouteSegmentPoint;
+struct RouteSegment;
+struct GpxPoint;
+struct GpxRouteApproximation;
+class RoutePlannerFrontEnd;
+
+class GpxMultiSegmentsApproximation {
+    // sync GpxMultiSegmentsApproximation.java
+    static constexpr int MAX_DEPTH_ROLLBACK = 500;
+    static constexpr double MIN_BRANCHING_DIST = 10;
+    static constexpr bool EAGER_ALGORITHM = false;
+    static constexpr bool PRIORITY_ALGORITHM = !EAGER_ALGORITHM;
+    static constexpr int ROUTE_POINTS = 12;
+    static constexpr int GPX_MAX = 30;
+    static constexpr bool DEBUG = false;
+
+public:
+    GpxMultiSegmentsApproximation(const SHARED_PTR<GpxRouteApproximation>& gctx,
```

Comment:
Add destructor for delete RouteSegmentAppr cache

Reply:
done